sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -832,27 +869,61 @@ def map_data_versions(
             self.unpause_snapshots(updated_prod_environment.snapshots, now_timestamp())
 
     def _snapshot_id_filter(
-        self, snapshot_ids: t.Iterable[SnapshotIdLike]
-    ) -> t.Union[exp.In, exp.Boolean]:
+        self, snapshot_ids: t.Iterable[SnapshotIdLike], fq_table_name: str
+    ) -> t.Union[exp.In, exp.Boolean, exp.Condition]:
         if not snapshot_ids:
             return exp.false()
-
-        return t.cast(exp.Tuple, exp.convert((exp.column("name"), exp.column("identifier")))).isin(
-            *[(snapshot_id.name, snapshot_id.identifier) for snapshot_id in snapshot_ids]
-        )
+        elif self.engine_adapter.SUPPORTS_TUPLE_IN:
```

Comment:
This branching seems too much, especially provided we use inheritance. Can we just override this method and `_snapshot_name_version_filter` in the `MSSQLEngineApdater` instead and get rid of this if / else?